### PR TITLE
dojson: allow multiple experiments on hep records

### PR DESCRIPTION
**Pull request by david-caro**
Signed-off-by: David Caro <david@dcaro.es>

### CODE REVIEW

File: inspirehep/dojson/hep/fields/bd6xx.py [hunk 1]
```diff
@@ -45,36 +47,54 @@ def field_categories2marc(self, key, value):
...
+            yield {
+                'record': get_record_ref(recid, 'experiments'),
+                'accelerator': acc_exp_data.get('a'),
+                'experiment': experiment_name,
+                'curated_relation': curated_relation
+            }
+
+    acc_exps_json = self.get('accelerator_experiments', [])
+    for acc_exp_data in force_force_list(acc_exps_data):
+        acc_exps_json.extend(_get_acc_exp_json(acc_exp_data))
```
Comment (jacquerie) [inline inspirehep/dojson/hep/fields/bd6xx.py]: This line is amazing to pronounce.

File: inspirehep/dojson/hep/fields/bd6xx.py [hunk 2]
```diff
@@ -45,36 +47,54 @@ def field_categories2marc(self, key, value):
...
+                recids = [
+                    int(recid) for recid
+                    in force_force_list(acc_exp_data.get('0'))
+                ]
+            except (TypeError, ValueError, AttributeError):
+                pass
+
+        experiment_names = force_force_list(acc_exp_data.get('e'))
+
+        for recid, experiment_name in izip_longest(recids, experiment_names):
```
Comment (jacquerie) [inline inspirehep/dojson/hep/fields/bd6xx.py]: Not convinced by this. We might zip together things that are not related with each other...

File: tests/unit/dojson/test_dojson_hep.py [hunk 3]
```diff
@@ -1249,13 +1249,209 @@ def test_field_categories(marcxml_to_json, json_to_marc):
...
-    assert (marcxml_to_json['accelerator_experiments'][0]['accelerator'] ==
-            json_to_marc['693'][0]['a'])
-    assert (marcxml_to_json['accelerator_experiments'][0]['experiment'] ==
-            json_to_marc['693'][0]['e'])
+    mock_get_record_ref.side_effect = \
+        lambda x, *_: x and 'mocked_record_%s' % x
+    mock_get_recid_from_ref.side_effect = \
+        lambda x, *_:  x and int(x.rsplit('_')[-1])
+
+
```
Comment (jacquerie) [inline tests/unit/dojson/test_dojson_hep.py]: Is the double newline intended here?

Comment (david-caro) [inline tests/unit/dojson/test_dojson_hep.py]: nop

File: inspirehep/dojson/hep/fields/bd6xx.py (hunk 1 above)
Comment (david-caro) [inline inspirehep/dojson/hep/fields/bd6xx.py]: writing the full 'accelerator_experiments...' felt too much, I can change if you find it unreadable

Comment (jacquerie) [inline inspirehep/dojson/hep/fields/bd6xx.py]: It was just a joke : )

Comment (david-caro) [inline inspirehep/dojson/hep/fields/bd6xx.py]: Quel pazzo ha rubato un pizzo prezioso con un pezzo di pizza in un pozzo!

**Comment by david-caro** [comment]
replaced by #1568
